data-prepper-plugins/common/build.gradle

```diff
@@ -3,6 +3,7 @@ plugins {
 }
 dependencies {
     compile project(':data-prepper-api')
+    compile project(':data-prepper-plugins:blocking-buffer')
```

Comment:
can you confirm why it needs to be here - Are we using this as part of any other common plugin ?

Reply:
Yes, the BufferFactoryTests and FileSourceTests are using the BlockingBuffer.

Reply:
I think if its just for tests - we should consider moving tests related BlockingBuffer to its specific project. And for other tests, we should consider using a DummyBuffer over this. So, that way if we update Plugin architecture to Service Loader interface, we can decouple these completely. Not a must but since we are moving them lets move them completely.

Reply:
@yadavcbala Thanks for the catch! Did not notice that. Yes, I agree that we should use DummyBuffer for BufferFactoryTests and FileSourceTests. There is already `TestBuffer` in the buffer test utlilities which should be applicable to FileSourceTests. Not sure if we can annotate that one for `BufferFactoryTests`.

Reply:
I agree, went ahead with using `TestBuffer` in `FileSourceTests` and discussed offline with @yadavcbala to proceed with annotating `TestBuffer` for usage in `BufferFactoryTests`.